modules/snooker/src/test/ts/browser/BarsTest.ts

```diff
@@ -0,0 +1,107 @@
+  const container = SugarElement.fromHtml('<div style="position: absolute; left: 0; top: 0; width: 500px"></div>');
+  const table = SugarElement.fromHtml<HTMLTableElement>(tableHtml);
+  const colgroupTable = SugarElement.fromHtml<HTMLTableElement>(colgroupTableHtml);
+  Insert.append(container, table);
+  Insert.append(container, colgroupTable);
+  Insert.append(SugarBody.body(), container);
+
+  const resizeAlwaysWire = ResizeWire.only(table, Fun.always);
+  const resizeCallbackWire = ResizeWire.only(table, isResizable);
+
+  // Test with resize allowed
+  Bars.refresh(resizeAlwaysWire, table);
+  assertdBarCounts(container, 1, 2);
+  Bars.destroy(resizeAlwaysWire);
+  assertdBarCounts(container, 0, 0);
+
+  // Test with resize false atrribute on table
+  setResizeAttr(container, 'table', 'false');
+  Bars.refresh(resizeCallbackWire, table);
+  assertdBarCounts(container, 0, 0);
+  Bars.destroy(resizeCallbackWire);
+  assertdBarCounts(container, 0, 0);
+  setResizeAttr(container, 'table', 'true');
+
+  // Test with resize false atrribute on tr
+  setResizeAttr(container, 'tr', 'false');
+  Bars.refresh(resizeCallbackWire, table);
+  assertdBarCounts(container, 0, 2);
+  Bars.destroy(resizeCallbackWire);
+  assertdBarCounts(container, 0, 0);
+  setResizeAttr(container, 'tr', 'true');
+
+  // Test with resize false atrribute on td
+  setResizeAttr(container, 'td', 'false');
+  Bars.refresh(resizeCallbackWire, table);
+  assertdBarCounts(container, 1, 1);
+  Bars.destroy(resizeCallbackWire);
+  assertdBarCounts(container, 0, 0);
+  setResizeAttr(container, 'td', 'true');
+
+  Remove.remove(container);
+});
+
+UnitTest.test('Bars.refresh and Bars.destroy with colgroup table', () => {
+  const container = SugarElement.fromHtml('<div style="position: absolute; left: 0; top: 0; width: 500px"></div>');
+  const table = SugarElement.fromHtml<HTMLTableElement>(colgroupTableHtml);
+  Insert.append(container, table);
+  Insert.append(SugarBody.body(), container);
+
+  const resizeAlwaysWire = ResizeWire.only(table, Fun.always);
+  const resizeCallbackWire = ResizeWire.only(table, isResizable);
+
+  // Test with resize allowed
+  Bars.refresh(resizeAlwaysWire, table);
+  assertdBarCounts(container, 1, 2);
+  Bars.destroy(resizeAlwaysWire);
+  assertdBarCounts(container, 0, 0);
+
+  // Test with resize false attribute on col
+  setResizeAttr(container, 'col', 'false');
+  Bars.refresh(resizeCallbackWire, table);
+  assertdBarCounts(container, 1, 1);
+  Bars.destroy(resizeCallbackWire);
+  assertdBarCounts(container, 0, 0);
+  setResizeAttr(container, 'col', 'true');
+
+  // Test with resize false attribute on td
+  setResizeAttr(container, 'td', 'false');
+  Bars.refresh(resizeCallbackWire, table);
+  assertdBarCounts(container, 1, 1);
+  Bars.destroy(resizeCallbackWire);
+  assertdBarCounts(container, 0, 0);
+  setResizeAttr(container, 'td', 'true');
+});
+
+// TODO: Test other functions from Bars.ts
```

Comment:
Do we have a Jira for this? If we're not planning to do this in this PR (which I'm fine with), could we please make sure there's a Jira for it?